pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTArrayAllocation.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+/* Generated By:JJTree: Do not edit this line. ASTAllocationExpression.java */
+
+package net.sourceforge.pmd.lang.java.ast;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+/**
+ * An array creation expression.
+ *
+ * <pre class="grammar">
+ *
+ * ArrayCreationExpression ::= "new" {@link ASTAnnotation TypeAnnotation}*
```

Comment:
For this one I originally went with `ArrayCreationExpression` (the JLS name), but I think it's too long and that `ArrayAllocation` is better than `ArrayCreation`. (I forgot to update the bnf here)

In fact, I'd like better `ArrayCreationExpr`. I think having an `Expr` suffix for all expression nodes would be nice for consistency, but that an `Expression` suffix like most nodes currently have is *too long* (especially in XPath expressions).

 IMHO `ArrayCreationExpr` would be better, but either all expression nodes have the suffix `Expr`, all have `Expression`, or we follow no convention. If we could rename the existing nodes from `*Expression` to `*Expr` I'd be inclined to use the shortened JLS name for this one. But I'm not sure breaking compatibility with existing nodes for a subjective reason like that is reasonable. Wdyt? Should all new nodes have the `*Expression` suffix to be consistent with the old ones? Or should we rename the old ones to `*Expr`, or follow no convention?

Reply:
Naming is difficult :) I'd wait for now with this decision, until we we have the AST changes done (and before we are fixing the rules). Having consistent naming is nice - and using the suffix "*Expr", sounds good for this case. We'd need to check, whether we could introduce other suffixes (Type, Name, Literal, ...)
However, since all the expression nodes will implement a common interface, this naming is not absolutely necessary.